src/FixedVector/TH.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module FixedVector.TH where
+
+import FixedVector.Type
+import Language.Haskell.TH
+import Numeric.Natural (Natural)
+
+-- import Language.Haskell.TH.Lib
+
+-- create a function that we would have written as follows:
+-- createVector<N> :: t -> t -> {N times} -> t -> FixedVector t <N>
+-- createVector<N> v1 v2 {...} v<N> = FixedVector { _data = [ v1, v2, {...}, v<N> ] }
+-- using Template Haskell: https://hackage.haskell.org/package/template-haskell-2.19.0.0/docs/Language-Haskell-TH.html
+-- TODO: don't depend on internals of FixedVector and instead use exposed constructors
+--       this might only be possible once we have e.g. createVectorUpToN, because then we can depend on the existence of these constructors
```

Comment:
```suggestion
```

@ners will TH recursively declare `createVector<N-1>` when we call `createVector N`?

Reply:
No. `createVector N` only declares `createVectorN`. That is why I suggest adding utilities such as `createVectorUpTo`.